apps/server/src/modules/account/repo/account.repo.ts

```diff
@@ -1,95 +1,145 @@
-import { AnyEntity, EntityName, Primary } from '@mikro-orm/core';
-import { ObjectId } from '@mikro-orm/mongodb';
+import { AnyEntity, EntityName, FilterQuery, Primary } from '@mikro-orm/core';
+import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
 import { Injectable } from '@nestjs/common';
-import { Account } from '@shared/domain/entity/account.entity';
 import { SortOrder } from '@shared/domain/interface';
-import { EntityId } from '@shared/domain/types';
-import { BaseRepo } from '@shared/repo/base.repo';
+import { Counted, EntityId } from '@shared/domain/types';
+import { AccountEntity } from '../entity/account.entity';
+import { AccountDoToEntityMapper } from './mapper/account-do-to-entity.mapper';
+import { Account } from '../domain/account';
+import { AccountEntityToDoMapper } from './mapper';
 
 @Injectable()
-export class AccountRepo extends BaseRepo<Account> {
+export class AccountRepo {
+	constructor(private readonly em: EntityManager) {}
+
 	get entityName() {
-		return Account;
+		return AccountEntity;
+	}
+
+	public async save(account: Account): Promise<Account> {
+		const saveEntity = AccountDoToEntityMapper.mapToEntity(account);
+		const existing = await this.em.findOne(AccountEntity, { id: account.id });
+
+		let saved: AccountEntity;
+		if (existing) {
+			saved = this.em.assign(existing, saveEntity);
```

Comment:
to note: mikro-orm has a problem when tryging to assign props which are private
you have to make sure you defined the type in @Propriety, otehrwise this can lead to error